src/MainWindow.vala

```diff
@@ -210,12 +211,14 @@ public class Mail.MainWindow : Hdy.ApplicationWindow {
         unowned Mail.Backend.Session session = Mail.Backend.Session.get_default ();
         session.account_added.connect (() => {
             placeholder_stack.visible_child = paned_end;
+            headerbar.can_compose = true;
```

Comment:
I don't think that the rest is necessary, the button state will be synced with the action sate
```suggestion
            get_action (ACTION_COMPOSE_MESSAGE).set_enabled (false);
```